systemFlake.nix

```diff
@@ -145,29 +124,35 @@ let
   );
 
 in
-otherArguments
-
-// eachSystem supportedSystems (system:
-  let
-    importChannel = name: value: import (patchChannel system value.input (value.patches or [ ])) {
-      inherit system;
-      overlays = sharedOverlays ++ (if (value ? overlaysBuilder) then (value.overlaysBuilder pkgs) else [ ]);
-      config = channelsConfig // (value.config or { });
-    };
-
-    pkgs = mapAttrs importChannel channels;
-
-    optional = check: optionalAttrs (check != null);
-  in
-  { inherit pkgs; }
-  // optional packagesBuilder { packages = packagesBuilder pkgs; }
-  // optional defaultPackageBuilder { defaultPackage = defaultPackageBuilder pkgs; }
-  // optional appsBuilder { apps = appsBuilder pkgs; }
-  // optional defaultAppBuilder { defaultApp = defaultAppBuilder pkgs; }
-  // optional devShellBuilder { devShell = devShellBuilder pkgs; }
-  // optional checksBuilder { checks = checksBuilder pkgs; }
-)
+mergeAny otherArguments (
+
+  eachSystem supportedSystems
+    (system:
+      let
+        importChannel = name: value: import (patchChannel system value.input (value.patches or [ ])) {
+          inherit system;
+          overlays = sharedOverlays ++ (if (value ? overlaysBuilder) then (value.overlaysBuilder pkgs) else [ ]);
+          config = channelsConfig // (value.config or { });
+        };
+
+        pkgs = mapAttrs importChannel channels;
+
+        mkOutput = output: builder: {
+          ${output} = otherArguments.${output}.${system} or { }
+          // optionalAttrs (args ? ${builder}) (args.${builder} pkgs);
+        };
+
+      in
+      { inherit pkgs; }
+      // mkOutput "packages" "packagesBuilder"
+      // mkOutput "defaultPackage" "defaultPackageBuilder"
+      // mkOutput "apps" "appsBuilder"
+      // mkOutput "defaultApp" "defaultAppBuilder"
+      // mkOutput "devShell" "devShellBuilder"
+      // mkOutput "checks" "checksBuilder"
```

Comment:
credit to @gytis-ivaskevicius for this idea. This looks really nice and I don't have to make mergeAny recursive.